File: server/src/graphQL/utils/getUser.ts
import findUserByID from '../../database/postgres/utils/findUserByID';
import GraphQLUser from '../types/GraphQLUser.type';

const getUser = async (id: string): Promise<GraphQLUser> => {
  const user = await findUserByID(id);
  if (user == null) {
    throw new Error('User does not exist.');
  }

  const { first_name: firstName, last_name: lastName, bio } = user;
  return {
    id,
    name: `${firstName} ${lastName}`,
    firstName,
    lastName,
    bio,
  };
};

export default getUser;

File: gin/utils/get_user.go
package utils

type User struct {
	ID   string `json:"id"`
	Name string `json:"name"`
}

func GetUser(id string) (user User, err error) {
	users := []User{
		{"0", "Hongbo"},
		{"1", "Jack"},
		{"2", "Rose"},
	}
	for _, u := range users {
		if u.ID == id {
			return u, nil
		}
	}
	return
}
